### PR TITLE
feature/endpoint-conflict

### PR DESCRIPTION
**Pull request by victorbalssa**

### CODE REVIEW

File: adback-solution-to-adblock/includes/class-ad-back-updator.php [hunk 1]
```diff
@@ -136,6 +136,25 @@ public static function isRewriteRouteEnabled()
         return (bool) get_option('permalink_structure');
     }
 
+    public static function isRewriteRulesConflictWithEndpoints($endpoint)
+    {
+        require_once( ABSPATH . 'wp-includes/class-wp-rewrite.php' );
```
Comment (nicolasThal) [inline adback-solution-to-adblock/includes/class-ad-back-updator.php]: why do you need to require the class `WP_Rewrite` ?

File: adback-solution-to-adblock/includes/class-ad-back-updator.php [hunk 2]
```diff
@@ -136,6 +136,25 @@ public static function isRewriteRouteEnabled()
...
+
+        if (!$rules = get_option('rewrite_rules')) {
```
Comment (nicolasThal) [inline adback-solution-to-adblock/includes/class-ad-back-updator.php]: Does it directly output an array or do you have to unserialize it ?

File: adback-solution-to-adblock/includes/class-ad-back-updator.php (hunk 1 above)
Comment (gignonje) [reply adback-solution-to-adblock/includes/class-ad-back-updator.php]: I'm not sure this has to be done like this.

Have a loot at WP_rewrite usage: https://codex.wordpress.org/Class_Reference/WP_Rewrite

```global $wp_rewrite;```  
```die(var_dump($wp_rewrite->rules)));```

File: adback-solution-to-adblock/includes/class-ad-back-updator.php (hunk 2 above)
Comment (victorbalssa) [reply adback-solution-to-adblock/includes/class-ad-back-updator.php]: `get_option()` return unserialized array yes

File: adback-solution-to-adblock/includes/class-ad-back-updator.php (hunk 1 above)
Comment (victorbalssa) [reply adback-solution-to-adblock/includes/class-ad-back-updator.php]: Yes I don't need this require anymore with `get_option()` function

**Pull request merged**